test/files.js

```diff
@@ -0,0 +1,138 @@
+/* eslint-env mocha */
+'use strict'
+
+const expect = require('chai').expect
+const bl = require('bl')
+const bs58 = require('bs58')
+const Readable = require('readable-stream')
+const isNode = require('detect-node')
+const path = require('path')
+
+let testfileBig
+
+if (isNode) {
+  const fs = require('fs')
+  testfileBig = fs.readFileSync(path.join(__dirname, './test-data/15mb.random'), { bufferSize: 128 })
+}
+
+module.exports = (common) => {
+  describe('.file', () => {
+    let ipfs
+
+    before((done) => {
+      common.setup((err, _ipfs) => {
+        expect(err).to.not.exist
+        ipfs = _ipfs
+        done()
+      })
+    })
+
+    after((done) => {
+      common.teardown(done)
+    })
+
+    describe('.cat', () => {
+      it('returns file stream', (done) => {
+        const hash = 'QmPZ9gcCEpqKTo6aq61g2nXGUhM4iCL3ewB6LDXZCtioEB'
+        ipfs.cat(hash, (err, file) => {
+          expect(err).to.not.exist
+          file.pipe(bl((err, bldata) => {
+            expect(err).to.not.exist
+            expect(bldata.toString()).to.contain('Check out some of the other files in this directory:')
+            done()
+          }))
+        })
+      })
+
+      // This fails on js-ipfs-api
```

Comment:
that's an old comment from before i made this pr to upgrade the js-ipfs-api to accept buffer inputs 

https://github.com/ipfs/js-ipfs-api/pull/280